pkg/ipcache/types/entries.go

```diff
@@ -20,12 +19,12 @@ func (s IPListEntrySlice) Swap(i, j int) {
 // Given that the same IP cannot map to more than one identity, no further
 // sorting is performed.
 func (s IPListEntrySlice) Less(i, j int) bool {
-	_, iNet, _ := net.ParseCIDR(*s[i].Cidr)
-	_, jNet, _ := net.ParseCIDR(*s[j].Cidr)
-	iPrefixSize, _ := iNet.Mask.Size()
-	jPrefixSize, _ := jNet.Mask.Size()
+	iNet, _ := netip.ParsePrefix(*s[i].Cidr)
+	jNet, _ := netip.ParsePrefix(*s[j].Cidr)
+	iPrefixSize := iNet.Bits()
+	jPrefixSize := jNet.Bits()
 	if iPrefixSize == jPrefixSize {
-		return bytes.Compare(iNet.IP, jNet.IP) < 0
+		return iNet.Addr().Compare(jNet.Addr()) < 0
```

Comment:
Would using netip.Addr.Less be functionally the same and a bit simpler?

Another bit that bothers me a bit is that the addresses need to be parsed repeatedly to be compared, would it be possible to store them in the slice in parsed form?

Reply:
Good call @jrajahalme! Commit c5c121e updates the PR to use the `Less()` method.